### PR TITLE
E2E: Integrate end-to-end workflows

### LINKED ISSUE #236: Integrate end-to-end workflows
**What would you like to be added**:
Integrate end-to-end workflows
**Why is this needed**:
Can better ensure the quality and stability of the software

### PR DESCRIPTION
**Pull request by lvtaococo**
#### What type of PR is this?
/kind e2e

#### What does this PR do?

Integrate end-to-end workflows
#### Which issue(s) does this PR fix?

Fixes #236 
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          fetch-depth: 0
+      - name: Run e2e test
+        run: ./hack/rune2e.sh
+      - name: Upload logs
+        uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: kosmos-e2e-logs-${{ github.run_id }}
+          path: ${{ github.workspace }}/e2e-test/logs-*
```
Comment (wuyingjun-lucky) [inline .github/workflows/e2e.yml]: Add a new line

Comment (lvtaococo) [reply .github/workflows/e2e.yml]: ok

**Pull request merged**